bin/interruptable.ts

```diff
@@ -0,0 +1,13 @@
+import { fork, join, monitor, Operation, Context } from 'effection';
+
+export function* interruptable(operation: Operation): Operation {
+  let context: Context = yield fork(operation);
+
+  yield monitor(({ ensure }) => {
+    let halt = () => context.halt();
+    process.on('SIGINT', halt);
+    ensure(() => process.off('SIGINT', halt));
+  });
+
+  return yield join(context);
+}
```

Comment:
I think mostly because it is used my our process entry points like `bin/start.ts`, `bin/watch.ts`, and `bin/parce-server.ts`. We can definitely move it elsewhere.